### ISSUE TITLE
Theme issues in Vaadin 14

### ISSUE THREAD

**Issue by heruan** [vaadin/flow#5642]
Upgrading from 13 to 14 alpha (currently `alpha3`) leads to issues with themes: at first app load or forced refresh, all is fine; navigating in app between views, some components lose their theme (i.e. Lumo).

For example, this is a grid after refresh:

<img width="454" alt="Screen Shot 2019-05-08 at 14 55 27" src="https://user-images.githubusercontent.com/851745/57376894-6504d680-71a1-11e9-93bd-4b6eefd34f99.png">

and this when navigating to the same view in-app:

<img width="444" alt="Screen Shot 2019-05-08 at 14 55 17" src="https://user-images.githubusercontent.com/851745/57376910-6df5a800-71a1-11e9-8560-ef8ad5d47976.png">

**Comment by heruan** [comment]
@pleku were you able to investigate on this? Would be great to have this fixed in next V14 pre-release!

**Comment by heruan** [comment]
This still affects `14.0.0.beta1`, quite unexpectedly since that means theming doesn't work in the first beta; I mean, does it? Is this confirmed by the team?

**Comment by pleku** [comment]
Sorry for the lack of response.
There seems to be some issues with the ordering of imports, effecting theming and components #5696.
We are looking into this.

**Comment by heruan** [comment]
Thank you @pleku for the update. This happens to me not only with Grid but also other components, e.g. it happens I have unstyled notifications and dialogs.

**Comment by pleku** [comment]
Ok. I'll be honest I'm not then 100 % sure it is covered by the refactoring and fixes that we have WIP for beta 2, targeted this mid-week.

I have to ask you to sit tight until beta2 and retest then. If it reproduces, then it would be nice to get a simple test case for us to replicate it easily.

**Comment by heruan** [comment]
Thank you @pleku for the feedback. Am I wrong if I say that right now V14 has theming broken for any component?

This is a demo project which replicates the issue: https://github.com/heruan/vaadin-issues

1. run with

```
mvn clean spring-boot:run -Dspring-boot.run.jvmArguments="-Dserver.port=9990 -Dvaadin.bowerMode=true"
```

2. go to http://localhost:9990

3. click the "Other" link.

You will see a button not style, clicking the button it will appear an unstyled notification; forcing a browser refresh, all styles load fine.

**Comment by heruan** [comment]
To simplify debugging, I've removed Spring from the sample project which can now be run simply with:

```
mvn clean jetty:run -DbowerMode=true
```

**Comment by Artur-** [comment]
The problem is that unthemed component versions are loaded on every second navigation... should be fixed by #5730

**Comment by heruan** [comment]
I heard rumors at DevDays about an altar in Turku's HQ where developers make offerings to @Artur- and, sometimes, they get a PR back.
